### PR TITLE
[backport/1.10] luajit: bump new version

### PR DESCRIPTION
**Pull request by igormunkin**
* ci: add LUAJIT_ENABLE_CHECKHOOK for exotic matrix
* ci: add ARM64 architecture to exotic testing
* ci: update action/checkout to v3
* Fix os.date() for wider libc strftime() compatibility.
* x86/x64: Fix loop realignment.
* ci: introduce workflow for exotic builds
* test: add test for `string.format('%c', 0)`
* ci: drop obsolete arguments for LuaJIT integration
* ci: stop using Ninja for integration testing

Part of #8069

NO_DOC=LuaJIT submodule bump
NO_TEST=LuaJIT submodule bump

### CODE REVIEW

**Review by Buristan — APPROVED** [review]
Hi, Igor!
Thanks for the patch!
LGTM!

File: changelogs/unreleased/gh-8069-luajit-fixes.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## bugfix/luajit
+
+Backported patches from vanilla LuaJIT trunk (gh-8069). In the scope of this
+activity, the following issues have been resolved:
+
+* Fixed loop realigment for dual-number mode
```
Comment (ylobankov) [inline changelogs/unreleased/gh-8069-luajit-fixes.md:6]: A dot at the end is missing :)

**Review by ylobankov — APPROVED** [review]
LGTM

File: changelogs/unreleased/gh-8069-luajit-fixes.md (hunk 1 above)
Comment (igormunkin) [reply changelogs/unreleased/gh-8069-luajit-fixes.md:6]: Oops, @kyukhin is too fast (and too furious)...

**Pull request merged**